site/topics/sorting/sorting.rst

```diff
@@ -38,7 +38,78 @@ Sorting Algorithms
     high-level idea, not the actual, literal implementations. This idea will be emphasized as we discuss the individual
     sorting algorithms below.
 
-   
+
+Selection Sort
```

Comment:
I always found this algorithm to be kinda a waste of time. The algorithm is basically identical to insertion sort, just getting the "next" element instead of placing the "current" element in the right order. Plus, it's worse in basically all regards to insertion sort in practice (unless your memory is spatially distant or like, a tape drive).

I'm not sure if bubble sort is any more intuitive, but it's quite a bit different from other sorts.

Reply:
Early on, it's more about having easily identifiable things they can recognize as having done before.

Reply:
I feel like I do insertion sort when sorting a shelf of books or something.

Reply:
It's always interesting how the sorting first plays out:
- Insertion and Selection are all ones they agree they have done before
- Bubble sort is ones they may or may not feel like they've done before, but nearly always, when asking the class for a sorting algo for the first time, they try to explain something nearly bubble sort. The usual hangup is the repeated scans.

Reply:
Except that one kid that apparently uses quicksort to sort all his stuff 🙄

Reply:
I use bogosort to cheat at cards.

Reply:
_pog_-o sort

Reply:
But, yeah. I'm not necessarily advocating bubble sort to start with. Insertion sort is more intuitive than that, and it's much more useful than selection sort (most standard library sorting algorithms are quicksort when array large, then use insertion sort when array small). Given the similarity to selection sort, I'm moreso saying just skip selection sort. Maybe show bubble sort as "yo, there's other comparison sorting algorithms" that don't use recursion.

101 doesn't _really_ go into sorting anyways. Going a little light makes sense to me.

Reply:
Idk ---- I like the possible progression of selection -> insertion 

1. Build a new list by always tacking on the next proper element to the end 
2. Now, instead, just add stuff to the new list by putting the things in their proper place within it 
3. ...Bubble is weird. 

I suppose the takeaway is:
- Think about a problem you all have experienced 
- See different algos for the same problem
- Think a bit about the algos

Reply:
I spent the day thinking about it and I still think including Selection Sort is worth while at this stage. I took your perspective and challenged the fact that _this is how I did it every other year_, but I still think it's worth showing the 3 given the big part of identifying a strategy they have definitely done in real life.

**However** , with that said, my current though process is to either (a) leave it as 3 distinct sections, or (b) have a section on Insertion Sort, but then have a subsection (or not even, perhaps just a note or something) on Selection Sort within the section. 

I'll let the ideas brew a little.

Reply:
All right. If you're convinced, I won't push. Questioning the status quo is good enough.